tsl/src/bgw_policy/job_api.c

```diff
@@ -288,8 +295,9 @@ job_alter(PG_FUNCTION_ARGS)
 	job = find_job(job_id, PG_ARGISNULL(0), if_exists);
 	if (job == NULL)
 		PG_RETURN_NULL();
-
-	ts_bgw_job_permission_check(job);
+	/* if this is the error retention job, skip the permission check and allow modifications */
+	if (!is_error_retention_job(job))
+		ts_bgw_job_permission_check(job);
```

Comment:
Not sure why that is necessary. Circumventing permission checks is always a risk, so would be good to understand why this is necessary.